### DISCUSSION TITLE
Questions about hrv_rsa()

### DISCUSSION THREAD

**Discussion by JU-THU** [neuropsychology/NeuroKit#652] [Q&A] [answered]
I have read the following paper and tried the process of hrv calculation.  https://doi.org/10.3390/s21123998
However, "nk.hrv" cannot output the result of rsa. Is anything wrong or any improvement with your code?
```python
#The code that I used
"import neurokit2 as nk
data_all = nk.data('https://raw.githubusercontent.com/neuropsychology/NeuroKit/dev/data/bio_resting_8min_200hz.json')
data_participant = data_all["S01"]
clean_signals, info = nk.bio_process(ecg=data_participant["ECG"],
rsp=data_participant["RSP"], sampling_rate=200)
info["ECG_R_Peaks"]
nk.hrv(info, sampling_rate=200, show=True)"
```

**Comment by DominiqueMakowski** [comment]
Can confirm that this code works on my machine. What's your error?

![image](https://user-images.githubusercontent.com/8875533/172272921-dbe83c9d-a0ac-4b22-9150-dc1840e38e25.png)

**Reply by JU-THU** [reply]
The result just showed the indices of hrv_time(), hrv_frequency() and hrv_nonlinear(), but no indices of hrv_rsa().

**Comment by DominiqueMakowski** [comment, marked as answer]
```python
import neurokit2 as nk

data = nk.data('https://raw.githubusercontent.com/neuropsychology/NeuroKit/dev/data/bio_resting_8min_200hz.json')["S01"]

clean_signals, info = nk.bio_process(ecg=data["ECG"], rsp=data["RSP"], sampling_rate=200)
nk.hrv_rsa(clean_signals, sampling_rate=200)
```
```
{'RSA_P2T_Mean': 0.16184523809523813,
 'RSA_P2T_Mean_log': -1.8211147207588794,
 'RSA_P2T_SD': 0.06775122709022904,
 'RSA_P2T_NoRSA': 1,
 'RSA_PorgesBohrer': -3.8425139100329213,
 'RSA_Gates_Mean': 8.339137462047258,
 'RSA_Gates_Mean_log': 2.120959789205375,
 'RSA_Gates_SD': 0.2133345212849366}
```

**Reply by JU-THU** [reply]
Thanks a lot. I got it. As the paper said "all the available indices can also be computed at once using hrv()", I thought hrv_rsa() was also contained in the function hrv(). Thank you for your patience again.
